### PR TITLE
collaps if x {} else { if .. } conditions

### PR DESCRIPTION
**Pull request by matthiaskrgr**
Fixes clippy::collapsible_else_if

reduces nesting of
````
if x {
 A
} else {
 if y {
  B
 } else {
  C
 }
}
````
to
````
if x {
 A
} else if y {
 B
} else {
 C
}
````

### CODE REVIEW

**Review by jyn514 — APPROVED** [review]
Rustdoc changes LGTM

File: compiler/rustc_codegen_ssa/src/back/link.rs [hunk 1]
```diff
@@ -1431,10 +1431,8 @@ fn add_late_link_args(
         if let Some(args) = sess.target.late_link_args_dynamic.get(&flavor) {
             cmd.args(args);
         }
-    } else {
-        if let Some(args) = sess.target.late_link_args_static.get(&flavor) {
-            cmd.args(args);
-        }
+    } else if let Some(args) = sess.target.late_link_args_static.get(&flavor) {
+        cmd.args(args);
     }
```
Comment (Mark-Simulacrum) [inline compiler/rustc_codegen_ssa/src/back/link.rs:1436]: FWIW I feel like this change is hiding intent - previously the code was pretty clearly handling the dynamic and non-dynamic cases, but now it is less obvious IMO, at least for me.

File: compiler/rustc_builtin_macros/src/asm.rs [hunk 2]
```diff
@@ -205,22 +205,20 @@ fn parse_args<'a>(
...
-        } else {
-            if !args.named_args.is_empty() || !args.reg_args.is_empty() {
-                let mut err = ecx.struct_span_err(
-                    span,
-                    "positional arguments cannot follow named arguments \
+        } else if !args.named_args.is_empty() || !args.reg_args.is_empty() {
+            let mut err = ecx.struct_span_err(
+                span,
+                "positional arguments cannot follow named arguments \
                      or explicit register arguments",
```
Comment (estebank) [inline compiler/rustc_builtin_macros/src/asm.rs:212]: Fix indentation

**Review by estebank — APPROVED** [review]
I'm slightly concerned about reshaping this hiding intent in some cases, but the code changes are all functionally the same.

What was the trigger to look into cleaning up this pattern in the codebase?

I'm r=me but see multiple places where I wish we had more comments.

**Comment by JohnCSimon** [comment]
Ping from triage
@matthiaskrgr  - can you please address the merge conflicts?  This already has approvals. Thank you.

@rustbot label: -S-waiting-on-review +S-waiting-on-author

**Comment by matthiaskrgr** [comment]
I'm closing this in regard to the criticism regarding readability.

**Pull request closed without merging**